### PR TITLE
Need guidance!

### PR DESCRIPTION
**Pull request by CharlieKenny**

### CODE REVIEW

File: .rspec [hunk 1]
```diff
@@ -0,0 +1,4 @@
+--color
+--require spec_helper
+--format documentation
+--require capybara/rspec
```
Comment (spike01) [inline .rspec:4]: This line would probably be better set up in your `spec_helper`

File: Gemfile [hunk 2]
```diff
@@ -1,3 +1,7 @@
 source 'https://rubygems.org'
 
 gem 'rspec'
+gem 'capybara'
+gem 'json'
+gem 'byebug'
+gem 'rspec-sinatra'
```
Comment (spike01) [inline Gemfile:7]: Where possible, try to separate your test and development dependencies in your Gemfile

File: Gemfile [hunk 3]
```diff
@@ -1,3 +1,7 @@
 source 'https://rubygems.org'
 
 gem 'rspec'
+gem 'capybara'
+gem 'json'
```
Comment (spike01) [inline Gemfile:5]: There is a JSON module as part of the Ruby standard library

File: lib/till.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
+require 'json'
+
+class Till
+
+  attr_accessor :customer_order
+  attr_accessor :receipt
+  attr_accessor :customer_bill
+  attr_accessor :tax
+  attr_accessor :total
```
Comment (spike01) [inline lib/till.rb:9]: Ok, but usually you'd probably not repeat `attr_accessor` over and over: you can write this on one line separating your accessors with commas. 

Also, needing this many public accessors is generally a code smell - do other classes really need to be able to read and write all these properties?

File: lib/till.rb [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  attr_reader :menu
+  
+  def initialize 
+    file = open("hipstercoffee.json")
+    json = file.read   
+    @menu = JSON.parse(json)
```
Comment (spike01) [inline lib/till.rb:15]: These 3 lines should be moved into a method

File: lib/till.rb [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+  attr_accessor :customer_order
+  attr_accessor :receipt
+  attr_accessor :customer_bill
+  attr_accessor :tax
+  attr_accessor :total
+  attr_reader :menu
+  
+  def initialize 
+    file = open("hipstercoffee.json")
```
Comment (spike01) [inline lib/till.rb:13]: This is a magic string - store it as a constant at the top of your class

File: lib/till.rb [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    @customer_bill
+  end
+
+
+  def totals
+    total_bill = @customer_bill.inject(0) { |result, element| result + element }
+    @total = total_bill
+    tax = @total * 0.0864
+    @tax = tax.round(2)
+  end
```
Comment (spike01) [inline lib/till.rb:44]: Where possible, try not to mutate state when calculating things like totals - this could just return the total from your `inject`

**Comment by spike01** [comment]
Hi Charlie, 

This is definitely moving in the right direction, but as you've said yourself there's some refactoring to be done. In general, always try to favour simplicity - if a method looks like it's doing too many unrelated things, it probably needs to be broken down into smaller pieces. Also, where possible, try to generate values dynamically - things like totals are a classic case for that, as they don't really need to be stored anywhere, since they can be calculated on the fly from whatever values you're trying to total. And as always, keep an eye on your formatting.

**Comment by CharlieKenny** [comment]
Hi @spike01 ,

Many thanks for the review, will get cracking on your pointers!

**Pull request closed without merging**